app/jobs/allocation_job.rb

```diff
@@ -0,0 +1,34 @@
+class AllocationJob < ApplicationJob
+  queue_as :default
+
+  def perform(allocation_batch_job)
+    school = allocation_batch_job.school
+    order_state = allocation_batch_job.order_state
+    current_allocation = school.std_device_allocation
+
+    current_allocation_value = current_allocation&.allocation || 0
+    new_allocation_value = current_allocation_value + allocation_batch_job.allocation_delta
+
+    current_cap_value = current_allocation&.cap || 0
```

Comment:
```suggestion
    current_cap_value = current_allocation&.raw_cap || 0
```

Reply:
I'll make the change although I can't actually make a failing test for this. I reckon as the state changes to either:
- `cannot_order` cap will get zeroed down to `devices_ordered`
- `can_order` cap gets set to the max allocation
So we could set `cap` to anything, I don't think it'd actually make a difference

Reply:
The test for this would be:

- Virtual pool with two schools at 10 devices each.
- Using this service object, update allocation for school A by 2.
- Does the virtual pool increase to 22 or 32? (It should be the former.)

Reply:
That line does set `current_cap_value` to the incorrect value with `cap` and correct value with `raw_cap`. However by the end of it all the outcome is the same and makes no difference hence why I can't get a test to fail

Reply:
> * pool with two schools at 10 devices each.
> * Using this service object, update allocation for school A by 2.
> * Does the virtual pool increase to 22 or 32? (It should be the former.)

`22` with either `cap` or `raw_cap`